x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.tsx

```diff
@@ -106,7 +106,9 @@ export function ErrorCountRuleType(props: Props) {
                 start,
                 end,
                 groupBy: params.groupBy,
-                searchConfiguration: JSON.stringify(params.searchConfiguration),
+                searchConfiguration: params.searchConfiguration?.query?.query
+                  ? JSON.stringify(params.searchConfiguration)
+                  : undefined,
```

Comment:
Nit: shouldn't the conditional just be `params.searchConfiguration`? 

```suggestion
                searchConfiguration: params.searchConfiguration
                  ? JSON.stringify(params.searchConfiguration)
                  : undefined,
```

Feel free to merge either way

Reply:
Actually... Shouldn't it be: 

```suggestion
                searchConfiguration: params.useKqlFilter
                  ? JSON.stringify(params.searchConfiguration)
                  : undefined,
```

?

Reply:
Once KQL filter is selected, `searchConfiguration` is set with empty `query` string. This was the issue that it was not adding normal filters even when KQL filter was turned off again because `searchConfiguration` now existed.

Reply:
Both would work (`useKqlFilter` and `searchConfiguration`) because when changing KQL toggle, all other filters and KQL filter are reset. Using `useKqlFilter` would be cleaner but I will let it be to not trigger new CI build.

Reply:
In that case the proper fix sounds like updating this:
https://github.com/elastic/kibana/blob/8ca63adbb7ab7ef3f26dc68b9db08620b3d714e4/x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.tsx#L250-L252

To: 
```
    setRuleParams('searchConfiguration', undefined);
```

Reply:
Initially I had set it to be `undefined`, but the Unified search bar is not shown in that case.

<img width="600" alt="Screenshot 2023-09-05 at 22 24 28" src="https://github.com/elastic/kibana/assets/69037875/50bbb96f-34bf-472c-bfdd-2ec2a22b8793">